frontend/src/components/Modal/IntroModal/IntroModal.test.js

```diff
@@ -95,40 +140,269 @@ describe("<IntroModal />", () => {
+        expect(wrapper.text()).toEqual("This username already exists");
+    });
+
+
+    it("should signinMessage reflect its state", () => {
+        const component = mount(introModal);
+        const introModalInstance = component.find(IntroModal.WrappedComponent).instance();
+
+        const openButton = component.find(".signin-open-button").hostNodes();
+        openButton.simulate("click");
+
+        introModalInstance.setState({ signinStatus: signinStatus.USER_NOT_EXIST });
+        const wrapper = component.find("#signin-message");
+        expect(wrapper.text()).toEqual("This user does not exist");
+
+        introModalInstance.setState({ signinStatus: signinStatus.WRONG_PW });
+        expect(wrapper.text()).toEqual("Wrong password");
+    });
+
+
+    it("should handle each result of signing up", () => {
+        stubInitialState = {
+            auth: {
+                signupStatus: signupStatus.WAITING,
+                signinStatus: signinStatus.NONE,
+            },
+        };
+        let component = mount(makeIntroModal(stubInitialState));
+        let openButton = component.find(".signup-open-button").hostNodes();
+        openButton.simulate("click");
+
+        let wrapper = component.find(".email-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_email@papersfeed.com" } });
+        wrapper = component.find(".username-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_username" } });
+        wrapper = component.find(".password-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_password" } });
+
+        let signupButton = component.find(".signup-button").hostNodes();
+        signupButton.simulate("click");
+
+        let introModalInstance = component.find(IntroModal.WrappedComponent).instance();
+        expect(introModalInstance.state.signupStatus).toBe(signupStatus.NONE);
+
+
+        stubInitialState = {
+            auth: {
+                signupStatus: signupStatus.SUCCESS,
+                signinStatus: signinStatus.NONE,
+            },
+        };
+        component = mount(makeIntroModal(stubInitialState));
+        openButton = component.find(".signup-open-button").hostNodes();
+        openButton.simulate("click");
+
+        wrapper = component.find(".email-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_email@papersfeed.com" } });
+        wrapper = component.find(".username-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_username" } });
+        wrapper = component.find(".password-input").hostNodes();
+        wrapper.simulate("change", { target: { value: "my_password" } });
+
+        signupButton = component.find(".signup-button").hostNodes();
+        signupButton.simulate("click");
+
+        introModalInstance = component.find(IntroModal.WrappedComponent).instance();
+        expect(introModalInstance.state.signupStatus).toBe(signupStatus.NONE);
```

Comment:
I think this happens because you change the state of introModal in line 49 of IntroModal.js.